### PR TITLE
Support paths in polyswarmd addresses

### PR DESCRIPTION
**Pull request by myhndl**

### CODE REVIEW

**Comment by myhndl** [comment]
I tested this with a mock engine, was able to connect to both `polyswarmd:31337` (hosted via orchestration) and `api.polyswarm.network/v1/default`

File: src/polyswarmclient/__init__.py [hunk 1]
```diff
@@ -234,6 +236,7 @@ def run(self, chains=None):
                     except (ValueError, aiohttp.ContentTypeError):
                         response = await raw_response.read() if raw_response else 'None'
                         logger.error('Received non-json response from polyswarmd: %s', response)
+                        logger.error('uri: %s', uri)
```
Comment (Rizato) [inline src/polyswarmclient/__init__.py]: Can you include this in the previous error log?

File: src/polyswarmclient/__init__.py [hunk 2]
```diff
@@ -330,7 +333,9 @@ def sign_transactions(self, transactions):
         if not is_valid_ipfs_uri(ipfs_uri):
             raise ValueError('Invalid IPFS URI')
 
-        uri = urljoin(self.polyswarmd_uri, '/artifacts/{0}/{1}'.format(ipfs_uri, index))
+        uri = self.polyswarmd_uri + '/artifacts/{0}/{1}'.format(ipfs_uri, index)
```
Comment (Rizato) [inline src/polyswarmclient/__init__.py]: Can we convert this to f string
```suggestion
        uri = f'{self.polyswarmd_uri}/artifacts/{ipfs_uri}/{index}'
```
https://github.com/polyswarm/polyswarm-client/blob/fe4deb4a52862c6f1ab91d3784249a9e8cb52911/src/worker/__init__.py#L103

File: src/polyswarmclient/__init__.py [hunk 3]
```diff
@@ -484,6 +491,7 @@ def get_artifacts(self, ipfs_uri, api_key=None):
                         except (ValueError, aiohttp.ContentTypeError):
                             response = await raw_response.read() if raw_response else 'None'
                             logger.error('Received non-json response from polyswarmd: %s', response)
+                            logger.error('uri: %s', uri)
```
Comment (Rizato) [inline src/polyswarmclient/__init__.py]: Combine error messages here

File: src/polyswarmclient/__init__.py [hunk 4]
```diff
@@ -434,7 +439,9 @@ def get_artifacts(self, ipfs_uri, api_key=None):
             (str): IPFS URI of the uploaded artifact
         """
 
-        uri = urljoin(self.polyswarmd_uri, '/artifacts')
+        uri = self.polyswarmd_uri + '/artifacts'
```
Comment (Rizato) [inline src/polyswarmclient/__init__.py]: Can we do f-string here too

File: src/polyswarmclient/__init__.py [hunk 5]
```diff
@@ -199,7 +200,8 @@ def run(self, chains=None):
         # Ensure we try at least once
         tries = max(tries, 1)
 
-        uri = urljoin(self.polyswarmd_uri, path)
+        uri = self.polyswarmd_uri + path
```
Comment (Rizato) [inline src/polyswarmclient/__init__.py]: f-string

File: src/polyswarmclient/__init__.py (hunk 1 above)
Comment (myhndl) [reply src/polyswarmclient/__init__.py]: done

**Comment by myhndl** [comment]
> I think like 300 could us an f-string too.

Done

File: src/polyswarmclient/__init__.py (hunk 3 above)
Comment (myhndl) [reply src/polyswarmclient/__init__.py]: done

File: src/polyswarmclient/__init__.py (hunk 5 above)
Comment (myhndl) [reply src/polyswarmclient/__init__.py]: done

File: src/polyswarmclient/__init__.py (hunk 4 above)
Comment (myhndl) [reply src/polyswarmclient/__init__.py]: done

**Pull request merged**